lib/context.js

```diff
@@ -54,6 +58,11 @@ function Context ({
   this.schemaErrorFormatter = schemaErrorFormatter || server[kSchemaErrorFormatter] || defaultSchemaErrorFormatter
   this[kRouteByFastify] = isFastify
 
+  this[kRequestValidateWeakMap] = new WeakMap()
+  this[kReplySerializeWeakMap] = new WeakMap()
```

Comment:
I don't think creating two WeakMaps _for every route_ is a good idea. Use an LRU that sits in the Fastify instance instead.

Reply:
Good point, but wouldn't that make the things a little bit more complicated as we'll need to bring a good default for customising the LRU cache size for cache eviction?

I support not building a `WeakMap` for every route as is possible the implementation is not going to use the serialize/validate feature for that specific route so doesn't have any sense to add it. Maybe we can go with a more conservative approach by lazy loading the WeakMap just if we detect usage, wdyt?